### PR TITLE
gh-115304: Add doc for initialize PyMutex as global variables

### PR DESCRIPTION
**Pull request by aisk**
* Issue: gh-115304

### CODE REVIEW

File: Include/internal/pycore_lock.h [hunk 1]
```diff
@@ -26,7 +26,7 @@ extern "C" {
 // 0b11: locked and has parked threads
 //
 // Typical initialization:
-//   PyMutex m = (PyMutex){0};
+//   PyMutex m = {0};
```
Comment (colesbury) [inline Include/internal/pycore_lock.h]: I don't think we should make this change. The comment is for variables that need explicit initialization, such as local variables and fields in heap allocated structures. Static variables are implicitly initialized to zero in C.

Comment (colesbury) [reply Include/internal/pycore_lock.h]: Maybe instead we can add an additional example of statically initializing the `PyMutex`?

Comment (aisk) [reply Include/internal/pycore_lock.h]: Thanks for the review! For local variables, should we remove the cast `(PyMutex)`? Seems it is needless in modern C.

Comment (colesbury) [reply Include/internal/pycore_lock.h]: Most commonly, we are initializing fields in a struct like:

```c
self->mutex = (PyMutex){0};
```

You can't write:

```c
self->mutex = {0};  // INVALID!
```

Comment (aisk) [reply Include/internal/pycore_lock.h]: Ah, I've got the point!

File: Include/internal/pycore_lock.h [hunk 2]
```diff
@@ -28,6 +28,9 @@ extern "C" {
 // Typical initialization:
 //   PyMutex m = (PyMutex){0};
 //
+// Or initialize as global variables:
+//   PyMutex m;
```
Comment (colesbury) [inline Include/internal/pycore_lock.h]: Let's tweak the example so it's clear that we're not declaring a local variable.

```suggestion
// Or initialize as global variables:
//   static PyMutex m;
```

**Review by colesbury — APPROVED** [review]
Thanks!

**Pull request merged**